src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -4725,9 +4730,39 @@ private void EmitMatchCharacterClass(string charClass, bool caseInsensitive, Loc
                 return;
             }
 
-            // Store the input character to a local so we can read it multiple times.
+            // All checks after this point require reading the input character multiple times,
+            // so we store it into a temporary local.
             Stloc(tempLocal);
 
+            // Next, if there's only 2 or 3 chars in the set (fairly common due to the sets we create for prefixess),
+            // it's cheaper and smaller to compare against each than it is to use a lookup table.
+            if (!invariant)
+            {
+                Span<char> setChars = stackalloc char[3];
+                int numChars = RegexCharClass.GetSetChars(charClass, setChars);
+                if (numChars > 0)
```

Comment:
aside, it seems to me that when `GetSetChars` returns 0, it doesn't mean "no chars match" so much as "not computed". Maybe -1 might have been better.

Reply:
It could mean either.  The net result is nothing could be given back.  I'm not sure -1 is better; then we'd just have to check <= 0.